starter/src/main/java/com/starter/fullstack/dao/InventoryDAO.java

```diff
@@ -60,32 +66,47 @@ public Inventory create(Inventory inventory) {
 
   /**
    * Retrieve Inventory.
+   *
    * @param id Inventory id to Retrieve.
    * @return Found Inventory.
    */
+
   public Optional<Inventory> retrieve(String id) {
-    // TODO
-    return Optional.empty();
+    Inventory inventory = this.mongoTemplate.findById(id, Inventory.class);
+    return Optional.ofNullable(inventory);
   }
 
   /**
    * Update Inventory.
-   * @param id Inventory id to Update.
+   *
+   * @param id        Inventory id to Update.
    * @param inventory Inventory to Update.
    * @return Updated Inventory.
    */
+
   public Optional<Inventory> update(String id, Inventory inventory) {
-    // TODO
+    Inventory existingInventory = this.mongoTemplate.findById(id, Inventory.class);
+    if (existingInventory != null) {
+      inventory.setId(id);
+      return Optional.of(this.mongoTemplate.save(inventory));
+    }
     return Optional.empty();
   }
 
   /**
    * Delete Inventory By Id.
-   * @param id Id of Inventory.
+   *
+   * @param id ID of Inventory.
    * @return Deleted Inventory.
    */
+
+  //TASK 3
   public Optional<Inventory> delete(String id) {
-    // TODO
+    Inventory inventory = this.mongoTemplate.findById(id, Inventory.class);
```

Comment:
No need to look up the inventory first then remove it. Instead of making two mongo calls there is another mongoTemplate method that you can use that will do both. You always want to make as little of mongo calls as possible.

Reply:
Used "findAndRemove" to resolve